test/timeouts.js

```diff
@@ -0,0 +1,147 @@
+const test = require('brittle')
+const { create, replicate } = require('./helpers')
+const b4a = require('b4a')
+
+test('get before timeout', async function (t) {
+  t.plan(1)
+
+  const core = await create()
+
+  const req = core.get(0, { timeout: 500 })
+
+  req.then((block) => {
+    t.alike(block, b4a.from('hi'))
+  }).catch((err) => {
+    t.fail(err.message)
+  })
+
+  await core.append('hi')
+})
+
+test('get after timeout', async function (t) {
+  t.plan(1)
+
+  const core = await create()
+
+  const req = core.get(0, { timeout: 500 })
+
+  req.then((block) => {
+    t.fail('should not have got block: ' + block)
+  }).catch((err) => {
+    t.is(err.code, 'REQUEST_TIMEOUT')
+  })
```

Comment:
Use await

Reply:
There was different behaviour between using await vs then+catch at one point, but with current PR not relevant anymore